### PR TITLE
Move users tables over to new schema migrations

### PR DESCRIPTION
**Pull request by nronas**
#### Summary
implements: https://mattermost.atlassian.net/browse/MM-33283

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33283

#### Release Note

```release-note
* Moves users gorp driven migrations to schema-based migrations
```

### CODE REVIEW

File: db/migrations/mysql/000012_create_users.up.sql [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    'ALTER TABLE Users DROP COLUMN LastPingAt;',
+    'SELECT 1'
+));
+
+PREPARE alterIfExists FROM @preparedStatement;
+EXECUTE alterIfExists;
+DEALLOCATE PREPARE alterIfExists;
+
+UPDATE Users SET Roles = 'system_user' WHERE Roles = '';
+UPDATE Users SET Roles = 'system_user system_admin' WHERE Roles = 'system_admin';
```
Comment (mgdelacroix) [inline db/migrations/mysql/000012_create_users.up.sql:62]: are these two potentially time expensive for big user tables?

File: db/migrations/mysql/000012_create_users.up.sql [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+DEALLOCATE PREPARE alterIfNotExists;
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'Users'
+        AND table_schema = DATABASE()
+        AND column_name = 'Position'
+    ) > 0,
+    'ALTER TABLE Users MODIFY Position varchar(128);',
```
Comment (mgdelacroix) [inline db/migrations/mysql/000012_create_users.up.sql:101]: shouldn't we check on the length of the column before making the change? Something along the lines of: https://github.com/mattermost/mattermost-server/pull/17463/files#diff-63823a10e7e7051a9fea834d002414805988bcd8d704793385ae2b81934b79a1R19

File: db/migrations/mysql/000012_create_users.up.sql [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+PREPARE alterIfExists FROM @preparedStatement;
+EXECUTE alterIfExists;
+DEALLOCATE PREPARE alterIfExists;
+
+UPDATE Users SET AuthData=LOWER(AuthData) WHERE AuthService = 'saml';
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT MIN(CHAR_LENGTH(Roles)) - MAX(CHAR_LENGTH(Roles))
```
Comment (mgdelacroix) [inline db/migrations/mysql/000012_create_users.up.sql:113]: same here, I think we can check if the length is different and modify if required

File: db/migrations/postgres/000012_create_users.up.sql [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    locale VARCHAR(5),
+    mfaactive boolean,
+    mfasecret VARCHAR(128)
+);
+
+ALTER TABLE users DROP COLUMN IF EXISTS lastactivityat;
+ALTER TABLE users DROP COLUMN IF EXISTS lastpingat;
+ALTER TABLE users ADD COLUMN IF NOT EXISTS position VARCHAR(128);
+ALTER TABLE users ADD COLUMN IF NOT EXISTS timezone VARCHAR(256) DEFAULT '{"automaticTimezone":"","manualTimezone":"","useAutomaticTimezone":"true"}';
+ALTER TABLE users ALTER COLUMN position TYPE VARCHAR(128);
```
Comment (mgdelacroix) [inline db/migrations/postgres/000012_create_users.up.sql:31]: is this a noop if the column has already this type? shouldn't we do something like this? https://github.com/mattermost/mattermost-server/pull/17463/files#diff-e01705889e6c5b9d74cfdf3879cd43b22aa506a5a89d11ee9fd0a8d3520cd07dR13

File: db/migrations/postgres/000012_create_users.up.sql [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+UPDATE Users SET AuthData=LOWER(AuthData) WHERE AuthService = 'saml';
+ALTER TABLE users ALTER COLUMN roles TYPE VARCHAR(256);
```
Comment (mgdelacroix) [inline db/migrations/postgres/000012_create_users.up.sql:33]: same here

**Review by streamer45 — APPROVED** [review]

**Comment by streamer45** [comment]
@isacikgoz This can be merged post conflict resolution.

**Pull request merged**